testing/extended/email_delivery_check.py

```
@@ -0,0 +1,207 @@
+    typ, data = conn.search(None, f'(SINCE "{since_str}")')
+    if typ != "OK" or not data or not data[0]:
+        return []
+    msg_ids = data[0].split()
+    out: list[Message] = []
+    for mid in msg_ids:
+        typ, msg_data = conn.fetch(mid, "(RFC822)")
+        if typ != "OK" or not msg_data:
+            continue
+        for part in msg_data:
+            if isinstance(part, tuple) and len(part) >= 2 and isinstance(part[1], (bytes, bytearray)):
+                out.append(email.message_from_bytes(bytes(part[1])))
```

Ruff (0.15.13)

[warning] Use `list.extend` to create a transformed list

(PERF401)